File: BrickBreakerADT/src/com/carlospinan/brickbreakerandroid/game/sprites/Paddle.java
package com.carlospinan.brickbreakerandroid.game.sprites;

import android.graphics.Bitmap;
import android.graphics.Canvas;

import com.carlospinan.brickbreakerandroid.configs.GameConfig;
import com.carlospinan.brickbreakerandroid.utils.Sprite;

/**
 * Created by mac on 14/07/13.
 */
public class Paddle extends Sprite {

    private static final float SPEED = 3.0f;

    public Paddle(Bitmap bitmap) {
        super(bitmap);
    }

    @Override
    public void draw(Canvas canvas) {
        super.draw(canvas);
    }

    public void move(float values[]) {
        float dx = values[0];
        setPosition(getX() + dx * SPEED, getY());

        if (getX() < getWidth() * 0.5f)
            setPosition(getWidth() * 0.5f, getY());

        if (getX() + getWidth() * 0.5f > GameConfig.SCREEN_WIDTH)
            setPosition(GameConfig.SCREEN_WIDTH - getWidth() * 0.5f, getY());
    }
}

File: BrickBreaker/app/src/main/java/com/carlospinan/brickbreaker/game/models/sprites/Paddle.kt
package com.carlospinan.brickbreaker.game.models.sprites

import android.graphics.Bitmap
import com.carlospinan.brickbreaker.game.SCREEN_WIDTH
import com.carlospinan.brickbreaker.game.models.Sprite
import kotlin.math.max
import kotlin.math.min

/**
 * @author Carlos Piñan
 */
private const val SPEED = 25.0f

class Paddle(bitmap: Bitmap) : Sprite(bitmap) {

    var direction: Float = 0.0f

    override fun update(dt: Float) {
        var deltaX = x + direction * SPEED * dt
        deltaX = max(deltaX, width / 2.0f)
        deltaX = min(deltaX, SCREEN_WIDTH - width / 2.0f)
        x = deltaX
    }

}